### PR TITLE
Add support for Terraform v0.12

### PR DESCRIPTION
**Pull request by aaron-lane**

### CODE REVIEW

**Comment by Jberlinsky** [comment]
@aaron-lane Could you kindly rebase this against `master` please?

**Comment by jeffmccune** [comment]
LGTM

File: test/fixtures/full/main.tf [hunk 1]
```diff
@@ -52,22 +52,22 @@ locals {
 }
 
 module "vpc" {
-  source       = "terraform-google-modules/network/google"
-  version      = "~> 0.4.0"
+  source = "git::https://github.com/terraform-google-modules/terraform-google-network.git?ref=master"
```
Comment (Jberlinsky) [inline test/fixtures/full/main.tf]: Can we pin this to a specific version? 0.12.x compatibility was introduced in https://github.com/terraform-google-modules/terraform-google-network/commit/7ba1b231e0662f4eaf264e5ad4a7e7ba977a40cd

File: examples/shared_vpc/main.tf [hunk 2]
```diff
@@ -39,33 +39,39 @@ provider "google-beta" {
...
-  source  = "terraform-google-modules/network/google"
-  version = "0.6.0"
+  # TODO: Switch to released version once
+  # https://github.com/terraform-google-modules/terraform-google-network/pull/47
+  # is merged and released.  This is here to fix the `Error: Unsupported block
+  # type` on the `triggers` block in network's main.tf file.
+  #
+  # source  = "terraform-google-modules/network/google"
+  # version = "0.8.0"
+  source = "git::https://github.com/terraform-google-modules/terraform-google-network.git?ref=master"
```
Comment (Jberlinsky) [inline examples/shared_vpc/main.tf]: Can we pin this to a specific version? 0.12.x compatibility was introduced in https://github.com/terraform-google-modules/terraform-google-network/commit/7ba1b231e0662f4eaf264e5ad4a7e7ba977a40cd

**Comment by Jberlinsky** [comment]
Sorry, race condition where this was merged while I was reviewing it. My suggestions are nit-picky -- would suggest making them follow-up issues for future addressing.

**Comment by DeMaBo** [comment]
There is occurred an error while using the updated version, it seems like an issue in the suite group resource. 

This is what my code looks like:

```
module "dev-project" {
  source  = "./modules/terraform-google-project-factory"

  name = var.project_dev
  random_project_id = "true"
  org_id = var.org_id
  usage_bucket_name = "dev-1-usage-report-bucket"
  billing_account = var.billing_account
}

```

The error message while executing plan and apply:
```
module.dev-project.module.gsuite_group.data.google_organization.org[0]: Refreshing state...

Error: Error reading Organization Not Found : {org_id}: Get https://cloudresourcemanager.googleapis.com/v1/organizations/{org_id}?alt=json&prettyPrint=false: oauth2: cannot fetch token: 400 Bad Request
Response: {
  "error": "invalid_grant",
  "error_description": "Bad Request"
}

  on modules/terraform-google-project-factory/modules/gsuite_group/main.tf line 26, in data "google_organization" "org":
  26: data "google_organization" "org" {


```

**Pull request merged**